analyzers/src/SonarAnalyzer.Common/Rules/ImplementSerializationMethodsCorrectlyBase.cs

```diff
@@ -71,6 +71,10 @@ public abstract class ImplementSerializationMethodsCorrectlyBase : SonarDiagnost
                 },
                 SymbolKind.Method);
 
+        private static bool HiddenByEditorBrowsableAttribute(IMethodSymbol methodSymbol) =>
+            methodSymbol.GetAttributes(KnownType.System_ComponentModel_EditorBrowsableAttribute)
+                .Any(x => x.ConstructorArguments.Any(a => 1.CompareTo(a.Value) == 0));
```

Comment:
This should be dejavalized to remove the indirection `1.CompareTo(a.Value) == 0`